src/server/hotkey_collector.cpp

```diff
@@ -16,14 +16,34 @@
 // under the License.
 
 #include "hotkey_collector.h"
+#include <dsn/dist/replication/replication_enums.h>
 
 namespace pegasus {
 namespace server {
 
-// TODO: (Tangyanzhao) implement these functions
+hotkey_collector::hotkey_collector(dsn::replication::hotkey_type::type hotkey_type,
+                                   dsn::replication::replica_base *r_base)
+    : replica_base(r_base), _state(hotkey_collector_state::STOPPED), _hotkey_type(hotkey_type)
+{
+}
+
 void hotkey_collector::handle_rpc(const dsn::replication::detect_hotkey_request &req,
                                   dsn::replication::detect_hotkey_response &resp)
 {
+    if (req.action == dsn::replication::detect_action::START) {
+        std::string err_hint;
+        if (start_detect(err_hint)) {
+            resp.err = dsn::ERR_OK;
```

Comment:
You can move this line at first in this function, then it's not needed to reassign it again and again.